### ISSUE TITLE
com.microsoft.playwright.TimeoutError: Error while

### ISSUE THREAD

**Issue by kairon007** [varabyte/kobweb#382]
*YOUR BUG DESCRIPTION HERE*

If relevant, specify:
* OS: LINUX
* Kobweb libs version: 0.14.1
* Kobweb CLI version: 0.9.13
---

To ensure this bug gets triaged / prioritized quickly, please...

While trying to use Google cloud run, there's a com.microsoft.playwright.TimeoutError thrown while navigating to "http://localhost:8080/admin/create?_kobwebIsExporting=true", waiting until "load" 

Use the default free cloud machine with ubuntu

Following is my dockerfile 
`#-----------------------------------------------------------------------------
# Variables shared across multiple stages (they need to be explicitly opted
# into each stage by being declaring there too, but their values need only be
# specified once).
ARG KOBWEB_APP_ROOT="site"

FROM eclipse-temurin:17 as java

FROM java as export

#-----------------------------------------------------------------------------
# Create an intermediate stage which builds and exports our site. In the
# final stage, we'll only extract what we need from this stage, saving a lot
# of space.

ENV KOBWEB_CLI_VERSION=0.9.13
ARG KOBWEB_APP_ROOT

ENV NODE_MAJOR=20

# Copy the project code to an arbitrary subdir so we can install stuff in the
# Docker container root without worrying about clobbering project files.
COPY . /project

# Update and install required OS packages to continue
# Note: Node install instructions from: https://github.com/nodesource/distributions#installation-instructions
# Note: Playwright is a system for running browsers, and here we use it to
# install Chromium.
RUN apt-get update \
    && apt-get install -y ca-certificates curl gnupg unzip wget \
    && mkdir -p /etc/apt/keyrings \
    && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
    && echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list \
    && apt-get update \
    && apt-get install -y nodejs \
    && npm init -y \
    && PLAYWRIGHT_DOWNLOAD_CONNECTION_TIMEOUT=120000 npx playwright install --with-deps chromium

# Fetch the latest version of the Kobweb CLI
RUN wget https://github.com/varabyte/kobweb-cli/releases/download/v${KOBWEB_CLI_VERSION}/kobweb-${KOBWEB_CLI_VERSION}.zip \
    && unzip kobweb-${KOBWEB_CLI_VERSION}.zip \
    && rm kobweb-${KOBWEB_CLI_VERSION}.zip

ENV PATH="/kobweb-${KOBWEB_CLI_VERSION}/bin:${PATH}"

WORKDIR /project/${KOBWEB_APP_ROOT}

# Decrease Gradle memory usage to avoid OOM situations in tight environments
# (many free Cloud tiers only give you 512M of RAM). The following amount
# should be more than enough to build and export our site.
RUN mkdir ~/.gradle && \
    echo "org.gradle.jvmargs=-Xmx256m" >> ~/.gradle/gradle.properties

RUN kobweb export --notty

#-----------------------------------------------------------------------------
# Create the final stage, which contains just enough bits to run the Kobweb
# server.
FROM java as run

ARG KOBWEB_APP_ROOT

COPY --from=export /project/${KOBWEB_APP_ROOT}/.kobweb .kobweb

ENTRYPOINT .kobweb/server/start.sh`

The dockerfile works well on render.com

**Comment by bitspittle** [comment]
I'm not sure this is related to Kobweb. How many times have you tried redeploying?

Personally I found working with Google Cloud Run (cheapest tier) very frustrating and temperamental and I often had to redeploy more than once to get something that stuck.

**Comment by kairon007** [comment]
> Unfortunately, I'm not sure there is anything I can do in the Kobweb codebase. How many times have you tried redeploying?
> 
> Personally I found working with Google Cloud Run (cheapest tier) very frustrating and temperamental and I often had to redeploy more than once to get something that stuck.

Tried it a lot of times using a tweaked docker and even the same one. When I tried deploying my container image via docker push it throws an error "site folder not found"

**Comment by bitspittle** [comment]
`site folder not found` feels like a misconfigured Docker file? Basically an earlier image should create a valid Kobweb site build and the last image should copy over just those bits. Do you have a top-level `site` folder in your project's root?

I often run Docker locally to verify that the image I wrote worked. Are you able to do that? That way we could verify the issue isn't the Dockerfile but is instead GCR somehow. (You said the file worked on render, though, right?)

**Comment by kairon007** [comment]
Yes absolutely the same docker worked on render and locally as well. I have
the site folder too. Somehow gcr is doing something weird.

**Comment by bitspittle** [comment]
That is so weird. 

Let me convert this issue to a discussion, so we can keep it going, because it doesn't feel right to close it.
